components/ws-daemon/pkg/cpulimit/cpulimit.go

```diff
@@ -218,8 +238,32 @@ type fixedLimiter struct {
 	FixedLimit Bandwidth
 }
 
-func (f fixedLimiter) Limit(budgetSpent CPUTime) (newLimit Bandwidth) {
-	return f.FixedLimit
+func (f fixedLimiter) Limit(wsh *WorkspaceHistory) (Bandwidth, error) {
+	return f.FixedLimit, nil
+}
+
+func AnnotationLimiter(annotation string) ResourceLimiter {
+	return annotationLimiter{
+		Annotation: annotation,
+	}
+}
+
+type annotationLimiter struct {
+	Annotation string
+}
+
+func (a annotationLimiter) Limit(wsh *WorkspaceHistory) (Bandwidth, error) {
+	value, ok := wsh.LastUpdate.Pod.Annotations[a.Annotation]
```

Comment:
Why are we storing the Pod and not only the annotations?

Reply:
It was just convenient to do and having the pod object around can be useful when we want to do limiting based on other attributes besides annotations.